common_test.go

```diff
@@ -42,3 +43,113 @@ func (s *MockGitUploadPackService) Fetch(*common.GitUploadPackRequest) (io.ReadC
 	r, _ := os.Open("formats/packfile/fixtures/git-fixture.ref-delta")
 	return r, nil
 }
+
+type SuiteCommon struct {
+	repos map[string]*Repository
+}
+
+var _ = Suite(&SuiteCommon{})
+
+var fixtureRepos = [...]struct {
+	url      string
+	packfile string
+}{
+	{"https://github.com/tyba/git-fixture.git", "formats/packfile/fixtures/git-fixture.ofs-delta"},
+}
+
+// create the repositories of the fixtures
+func (s *SuiteCommon) SetUpSuite(c *C) {
+	s.repos = make(map[string]*Repository, 0)
+	for _, fixRepo := range fixtureRepos {
+		s.repos[fixRepo.url] = NewPlainRepository()
+
+		d, err := os.Open(fixRepo.packfile)
+		defer func() {
+			c.Assert(d.Close(), IsNil)
+		}()
+		c.Assert(err, IsNil)
+
+		r := packfile.NewReader(d)
+		r.Format = packfile.OFSDeltaFormat // TODO: how to know the format of a pack file ahead of time?
+
+		_, err = r.Read(s.repos[fixRepo.url].Storage)
+		c.Assert(err, IsNil)
+	}
+}
+
+var countLinesTests = [...]struct {
+	i string // the string we want to count lines from
+	e int    // the expected number of lines in i
+}{
+	{"", 0},
+	{"a", 1},
+	{"a\n", 1},
+	{"a\nb", 2},
+	{"a\nb\n", 2},
+	{"a\nb\nc", 3},
+	{"a\nb\nc\n", 3},
```

Comment:
Yes, we want all the lines, even empty lines, as empty lines are part of some languages syntax.